src/components/ChatLog.jsx

```diff
@@ -2,24 +2,25 @@ import ChatEntry from './ChatEntry.jsx';
 import PropTypes from 'prop-types';
 import './ChatLog.css';
 
-const ChatLog = ({ entries, onToggleLike }) => {
+const ChatLog = ({ entries, onToggleLike = () => {} }) => {
 	return (
 		<div className="chat-log">
 			{entries.map((entry) => (
 				<ChatEntry
           key={entry.id}
+          id={entry.id} // Pass the id directly
           sender={entry.sender}
           body={entry.body}
           timeStamp={entry.timeStamp}
           liked={entry.liked}
-          onLikeToggle={() => onToggleLike(entry.id)}
+          onLikeToggle={onToggleLike} // Pass the function directly
```

Comment:
👍 We can drill the props down to the individual `ChatEntry` components rather than building a new function here. `ChatEntry` still makes its own per-component function `handleLikeClick`, so we're not doing this out of concern for the resources being consumed by the function, but rather fomr the perspective of which component should know how to wire up the function. In this case, `ChatLog`'s responsibility is passing data through to the individual entries, and the entries have the responsibility of knowing how to use the values.